src/core/hle/kernel/kernel.h

```diff
@@ -331,6 +333,15 @@ class KernelSystem {
 
     Core::Timing& timing;
 
+    /// Sleep main thread of the first ever launched non-sysmodule process.
+    void SetAppMainThreadExtendedSleep(bool requires_sleep) {
+        main_thread_extended_sleep = requires_sleep;
+    }
+
+    bool GetAppMainThreadExtendedSleep() {
+        return main_thread_extended_sleep;
```

Comment:
```suggestion
        return std::exchange(main_thread_extended_sleep, false);
```
Since the flag is always reset after being accessed

Reply:
I don't like this, the flag is always reset right now, but it may change in the future for any reason. A getter shouldn't be modifying the state, in fact I made it const.